packages/catalog-process/src/utilities/errorMappers.ts

```diff
@@ -355,3 +355,12 @@ export const rejectDelegatedEServiceDescriptorErrorMapper = (
     .with("eServiceDescriptorNotFound", () => HTTP_STATUS_NOT_FOUND)
     .with("operationForbidden", () => HTTP_STATUS_FORBIDDEN)
     .otherwise(() => HTTP_STATUS_INTERNAL_SERVER_ERROR);
+
+export const upgradeEServiceInstanceErrorMapper = (
+  error: ApiError<ErrorCodes>
+): number =>
+  match(error.code)
+    .with("eServiceNotFound", () => HTTP_STATUS_NOT_FOUND)
+    .with("eServiceDescriptorNotFound", () => HTTP_STATUS_NOT_FOUND)
+    .with("operationForbidden", () => HTTP_STATUS_FORBIDDEN)
+    .otherwise(() => HTTP_STATUS_INTERNAL_SERVER_ERROR);
```

Comment:
Missing `eserviceTemplateNotFound`, `eServiceNotAnInstance` and `eServiceAlreadyUpgraded` errors